apps/examples/logs_dynamic/logs_dynamic_main.c

```diff
@@ -0,0 +1,44 @@
+/****************************************************************************
+ *
+ * Copyright 2022 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <tinyara/serial/tioctl.h>
+#include <fcntl.h>
+#include<sys/ioctl.h>
+
+
+/****************************************************************************
+ * hello_main
+ ****************************************************************************/
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int logs_dynamic_main(int argc, char *argv[])
+#endif
+{
+	int fd = open("/dev/console", O_RDWR);
```

Comment:
So, you want user reveal an amount of logs from given size of buffer when user wants it, am I right?
Then why don't expand procfs instead of this approach?
Then developer can uses it very easily by tash command.

Reply:
Hi Mr. Kwon, currently we don't know how the network module is going to handle the stored buffer list. I have added a temporarily ioctl call just to verify the integrity of the saved logs. The user will not use the stored logs. So, I am not planning to add this to procfs.

Reply:
if buffer should be shared to network module, than procfs is best option.
That module can read data as a streaming by 'read'